aten/src/ATen/native/Loss.cpp

```diff
@@ -340,6 +342,49 @@ Tensor smooth_l1_loss_backward(const Tensor& grad_output, const Tensor& input, c
   return at::smooth_l1_loss_backward_out(grad_input, grad_output, input, target, reduction, beta);
 }
 
+Tensor huber_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
+  TORCH_CHECK(beta > 0, "huber_loss does not support non-positive values for beta.")
+  Tensor loss;
+  auto iter = TensorIterator::binary_op(loss, input, target);
+  huber_stub(iter.device_type(), iter, beta);
+  return apply_loss_reduction(iter.output(), reduction);
+}
+
+Tensor& huber_loss_out(const Tensor& input, const Tensor& target, int64_t reduction, double beta, Tensor& result) {
+  TORCH_CHECK(beta > 0, "huber_loss does not support non-positive values for beta.")
+  if (reduction != Reduction::None) {
```

Comment:
Does the reduction parameter need to be validated? This has an "else" clause, which will catch any value

Reply:
The reduction parameters are not validated anywhere within this file, but they are validated higher up `functional.py`.